epoxy-processor/src/main/java/com/airbnb/epoxy/ClassToGenerateInfo.java

```diff
@@ -203,5 +225,33 @@ public MethodInfo(String name, Set<Modifier> modifiers, List<ParameterSpec> para
       this.modifiers = modifiers;
       this.params = params;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
great! I like this approach of using equals to prevent duplicates. Since this class is getting large what do you think of making `MethodInfo`  top level class? We could also maybe move `collectMethodsReturningClassType` into that class as a static method to build a list of methods based on a class type